File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> sol_map = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (sol_map.containsKey(complement)) {
                return new int[] {sol_map.get(complement), i};
            }
            sol_map.put(nums[i], i);
        }
        return new int[] {};
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums) - 1
        while start <= end:
            midpoint = (start+end)//2
            if nums[midpoint] == target:
                return midpoint
            if nums[midpoint] < target:
                start = midpoint + 1
            else: 
                end = midpoint - 1
        return -1